### PR TITLE
EmailBundle - removing deprecated code

### PR DESCRIPTION
**Pull request by escopecz**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/7998
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Test all forms related to emails.
3. Including email related actions in campaign, form and point actions.

#### List deprecations along with the new alternative:
In https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

#### List backwards compatibility breaks:
In https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

### CODE REVIEW

**Comment by dongilbert** [comment]
Some conflicts need resolving

File: app/bundles/EmailBundle/Form/Type/AbTestPropertiesType.php [hunk 1]
```diff
@@ -35,7 +32,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
     /**
      * {@inheritdoc}
      */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setOptional([
```
Comment (anton-vlasenko) [inline app/bundles/EmailBundle/Form/Type/AbTestPropertiesType.php]: `setOptional` method is deprecated. Please use `setDefined` method instead.

File: app/bundles/EmailBundle/Form/Type/EmailOpenType.php [hunk 2]
```diff
@@ -45,21 +42,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
     }
 
     /**
-     * @param OptionsResolverInterface $resolver
+     * @param OptionsResolver $resolver
      */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setOptional(['list_options']);
```
Comment (anton-vlasenko) [inline app/bundles/EmailBundle/Form/Type/EmailOpenType.php]: `setOptional` method is deprecated. Please use `setDefined` method instead.

File: app/bundles/EmailBundle/Form/Type/AbTestPropertiesType.php (hunk 1 above)
Comment (escopecz) [reply app/bundles/EmailBundle/Form/Type/AbTestPropertiesType.php]: I don't see it documented in https://github.com/symfony/symfony/blob/3.0/UPGRADE-3.0.md#form nor in https://docs.google.com/document/d/1lHNK8hHQtvyWlz9OtfAoLupDVb5-QunQjq-HAGlAsS4/edit#

I will add it to our Gdoc

**Comment by escopecz** [comment]
> Shouldn't we flip keys and values for ChoiceType type and use 'choices_as_values' option? (actually you showed me this)

Yes, it should. I must have forgot that step here. Thanks for noticing. Ready for second round.

**Comment by escopecz** [comment]
@anton-vlasenko thanks! Fixed.

**Comment by escopecz** [comment]
I will check the VariantType once my laptop is fixed.

We are not dealing with controllers in M3 and there is another issue to refactor the transports so those won't be refactored in this PR.

**Comment by anton-vlasenko** [comment]
> there is another issue to refactor the transports so those won't be refactored in this PR.

@escopecz Thank you for explaining to me. I haven't noticed these tickets.

I just don't understand how the controllers are going to work if we don't refactor them. But If you say that they should not be refactored now I'm fine with that. I'm just curious what our plan is.
2. I'd like to ask what to do with the `\Mautic\EmailBundle\EmailEvents::ON_CAMPAIGN_TRIGGER_ACTION`. It's deprecated and it's not in the controller and/or transport.

**Comment by escopecz** [comment]
@anton-vlasenko I fixed the VariantType and created issue for the campaign batch deprecation https://github.com/mautic/mautic/issues/8191

**Comment by escopecz** [comment]
@anton-vlasenko here is the issue: https://github.com/mautic/mautic/issues/8060

Just to be clear what the current goal is. Let's make Mautic work on Symfony 3.4. If there will be time to refactor more we can do that. Removing MauticFactory is secondary goal. Let's focus on the primary goal.

**Comment by anton-vlasenko** [comment]
@escopecz Could you please merge it?

**Pull request merged**